### PR TITLE
drivers: adc : Add STM32G4X adc support

### PR DESCRIPTION
**Pull request by paoloteti**
Add adc0 and adc1 nodes and related driver support common to all STM32G4X SoC series.

### CODE REVIEW

**Comment by paoloteti** [comment]
recheck

File: dts/arm/st/g4/stm32g4.dtsi [hunk 1]
```diff
@@ -27,6 +27,26 @@
 	};
 
 	soc {
+		adc0: adc@50000000 {
```
Comment (erwango) [inline dts/arm/st/g4/stm32g4.dtsi:30]: Can you add a comment stating that support of both adc instances running in parallel is not supported right now?

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by erwango** [comment]
@galak do you know what is failing ?

File: dts/arm/st/g4/stm32g4.dtsi (hunk 1 above)
Comment (paoloteti) [reply dts/arm/st/g4/stm32g4.dtsi:33]: Done.

**Review by erwango — APPROVED** [review]

**Pull request merged**